Assets/Countly/Plugins/CountlySDK/Services/UserProfile.cs

```diff
@@ -0,0 +1,682 @@
+                }
+            }
+            CustomMods[truncatedKey] = ob;
+        } catch (Exception e) {
+            Log.Error($"[UserProfile][ModifyCustomData] An exception occured during modifying the custom data. Exception: {e}");
+        }
+    }
+
+    private void ClearInternal()
+    {
+        Log.Debug("[UserProfile][ClearInternal] Start");
+
+        Name = null;
+        Username = null;
+        Email = null;
+        Organization = null;
+        Phone = null;
+        PictureUrl = null;
+        Gender = null;
+        BirthYear = 0;
+        CustomDataProperties = new Dictionary<string, object>();
+        CustomMods = new Dictionary<string, JObject>();
+    }
+
+    private string ConvertToJSON()
+    {
+        JObject json = new JObject();
+
+        try {
+            if (Name != null) {
+                if (Name == "") {
+                    json.Add(NAME_KEY, null);
+                } else {
+                    json.Add(NAME_KEY, Name);
+                }
+            }
+
+            if (Username != null) {
+                if (Username == "") {
+                    json.Add(USERNAME_KEY, null);
+                } else {
+                    json.Add(USERNAME_KEY, Username);
+                }
+            }
+
+            if (Email != null) {
+                if (Email == "") {
+                    json.Add(EMAIL_KEY, null);
+                } else {
+                    json.Add(EMAIL_KEY, Email);
+                }
+            }
+
+            if (Organization != null) {
+                if (Organization == "") {
+                    json.Add(ORG_KEY, null);
+                } else {
+                    json.Add(ORG_KEY, Organization);
+                }
+            }
+
+            if (Phone != null) {
+                if (Phone == "") {
+                    json.Add(PHONE_KEY, null);
+                } else {
+                    json.Add(PHONE_KEY, Phone);
+                }
+            }
+
+            if (PictureUrl != null) {
+                if (PictureUrl == "") {
+                    json.Add(PICTURE_KEY, null);
+                } else if (utils.IsPictureValid(PictureUrl)) {
+                    json.Add(PICTURE_KEY, PictureUrl);
+                }
+            }
+
+            if (Gender != null) {
+                if (Gender == "") {
+                    json.Add(GENDER_KEY, null);
+                } else {
+                    json.Add(GENDER_KEY, Gender);
+                }
+            }
+
+            if (BirthYear != 0) {
+                if (BirthYear > 0) {
+                    json.Add(BYEAR_KEY, BirthYear);
+                } else {
+                    json.Add(BYEAR_KEY, null);
+                }
+            }
+
+            JObject ob = null;
+
+            if (CustomDataProperties.Count > 0) {
+                utils.TruncateSegmentationValues(CustomDataProperties, config.GetMaxSegmentationValues(), "[UserProfile][ConvertToJSON]", Log);

```

Comment:
btw log structure is not same as [SDK dev guide](https://support.countly.com/hc/en-us/articles/360037753291-SDK-development-guide#01H822TC8FN7N2AHT7C6TJKR95)
afaik it should be
[CLASS] functionName, LOG

Reply:
oh. didn't enjoy seeing this one 😄